### PR TITLE
Add Security section in elasticsearch.yml

### PR DESCRIPTION
**Pull request by bytebilly**
Adds a new "Security" section in the default elasticsearch.yml file to warn users that security features are disabled and linking to documentation.

The goal is to raise awareness in users that open the configuration file without following the getting started guide.

### CODE REVIEW

File: distribution/src/config/elasticsearch.yml [hunk 1]
```diff
@@ -80,3 +80,16 @@
 # Require explicit names when deleting indices:
 #
 #action.destructive_requires_name: true
+#
+# ---------------------------------- Security ----------------------------------
+#
+#                                 *** WARNING ***
+#
+# By default Elasticsearch doesn’t have security features enabled.
+# This means that users don’t have to provide a password and can get full access
```
Comment (jkakavas) [inline distribution/src/config/elasticsearch.yml:89]:
```suggestion
# This means that users don’t have to provide credentials and can get full access
```

**Review by jkakavas — APPROVED** [review]
LGTM. I'm not sure about the wording but I don't think we'll ever be and I don't have any concrete suggestions. I asked for a review from Adam too before we merge this

**Review by lockewritesdocs — APPROVED** [review]
Added a few minor edits, but LGTM. I think that being general is the best approach for this section.

File: distribution/src/config/elasticsearch.yml [hunk 2]
```diff
@@ -80,3 +80,16 @@
 # Require explicit names when deleting indices:
 #
 #action.destructive_requires_name: true
+#
+# ---------------------------------- Security ----------------------------------
+#
+#                                 *** WARNING ***
+#
+# Elasticsearch security features are disabled by default.
```
Comment (tvernum) [inline distribution/src/config/elasticsearch.yml:88]: This isn't true for paid (Gold+ licenses) and I wonder whether the message might scare some people.
I don't have a good suggestion, but if we can come up with something that still makes a strong statement with a little more accuracy.

Maybe

```suggestion
# Elasticsearch security features are not enabled by default.
# These features are free, but require configuration changes to enable them.
```

Comment (lockewritesdocs) [reply distribution/src/config/elasticsearch.yml:88]: I'm 👍 to this change. I think that brevity wins here.

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

**Pull request merged**